sphinx_immaterial/kbd_keys.py

```diff
@@ -0,0 +1,112 @@
+from typing import List, Tuple
```

Comment:
Sphinx already has a `kbd` role:
https://www.sphinx-doc.org/en/master/usage/restructuredtext/roles.html#other-semantic-markup

Reply:
It does. But the pymdownx ext allows for greater customization. That's why I chose `keys` as the rule name.

Reply:
There might be an instance where a user doesn't want to take advantage of the `keys_map` config option (which by default includes many aliases for English words/acronyms). That's where the original kbd role could be useful. Of course, if they don't want to use the `keys_map` at all, then they could simply not install pymdownx. But that is a global approach (using `kbd` role vs `keys` role would be context specific).

Reply:
It seems kind of unlikely that someone would want to display key bindings in two different ways in one project, though.  If they just want to avoid a particular alias, perhaps there can just be a way for them to modify it from `conf.py`?

Reply:
I haven't tried it, but theoretically one could try to override a default alias...
```py
keys_map = {"delete": "Delete"} # default for this is "Del"
```

---
If that works, then I'll happily override the kbd role.

Reply:
Tried overriding the `delete` alias in `keys_map`... It worked, but the theme CSS is still using the CSS classes to add symbols before/after certain keys, 

`:keys:`delete`` results in  ![image](https://user-images.githubusercontent.com/14963867/163066042-7267094a-ddd1-4a0c-99d9-cf01e8c981e4.png)

Using the `:kbd:`Delete`` results in ![image](https://user-images.githubusercontent.com/14963867/163066244-3fa7e74f-4d83-4a00-826a-cad2816c48ef.png)

Reply:
Just to re-cap this thread, I won't be overriding the original `kbd` role unless there's a way to allow the user to not use the theme's CSS (related to the `pymdownx.keys` ext).

Technically, there are several ways to do this globally (user could set `keys_class = <anything but "keys">` in conf.py), but no way (that I know of) to do this per inline parsing.

Reply:
I'd also like to point out that listing CLI options in RST ends up generating `<kbd>` elements. See the last row [in this table](https://sphinx-immaterial.readthedocs.io/en/latest/rst-cheatsheet/rst-cheatsheet.html#lists). But, that's different from using the kbd role (syntax must start with a `-` or `/`).

Reply:
Hmm --- yeah I noticed in the HTML spec also that there seemed to be multiple different interpretations of the `<kbd>` element --- sometimes as an individual key, other times just to indicate some form of keyboard input.

I'm not sure what we should do about that.

Reply:
My initial instinct is do nothing. The immediate question is: Do you still want to override the `kbd` role knowing that would invoke theme-specific CSS (which was solely designed for `pymdownx.keys`)?

If you decide that the CLI args list shouldn't use the same CSS for kbd role, then I guess we could override the CSS on our end - its not a problem upstream because MD doesn't have a native syntax for that scenario like RST does.

Reply:
Yes I agree --- let's just do nothing for now.